test/shared_test/test_ortops_hfberttokenizer.cc

```diff
@@ -141,3 +141,104 @@ TEST(hf_bert_tokenizer_opertor, test_default) {
     TestInference(*ort_env, model_path.c_str(), inputs[i], outputs[i]);
   }
 }
+
+TEST(hf_bert_tokenizer_operator, test_optional_output) {
+  auto ort_env = std::make_unique<Ort::Env>(ORT_LOGGING_LEVEL_WARNING, "Default");
+
+  const std::filesystem::path model_path = "data/test_hf_bert_tokenizer_optional_output.onnx";
+
+  const std::vector<int64_t> input_dims{2};
+  const std::string context1 =
+      "John is a 10 year old boy. "
+      "He is the son of Robert Smith. "
+      "Elizabeth Davis is Robert's wife. "
+      "She teaches at UC Berkeley. "
+      "Sophia Smith is Elizabeth's daughter. "
+      "She studies at UC Davis";
+  const std::string context2 =
+      "My name is John."
+      "I live in San Jose, California."
+      "Rob is my friend."
+      "He lives in Seattle, Washington.";
+
+  std::vector<std::vector<TestValue>> inputs = {
+      {TestValue{"text", {"Who is John's sister?", context1}, input_dims}},
+      {TestValue{"text", {"Where does sophia study?", context1}, input_dims}},
+      {TestValue{"text", {"Who is John's mom?", context1}, input_dims}},
+      {TestValue{"text", {"Where does John's father's wife teach?", context1}, input_dims}},
+      {TestValue{"text", {"Who is John's father's wife's daughter's brother?", context1}, input_dims}},
+      {TestValue{"text", {"Who is John's friend?", context2}, input_dims}},
+      {TestValue{"text", {"Where does John's friend live?", context2}, input_dims}},
+      {TestValue{"text", {"Which state does John's friend live?", context2}, input_dims}}};
+
+  std::vector<std::vector<TestValue>> outputs = {
+      {TestValue{
+           "input_ids",
+           std::vector<int64_t>{101, 2040, 2003, 2198, 1005, 1055, 2905, 1029, 102, 2198, 2003, 1037, 2184, 2095, 2214,
+                                2879, 1012, 2002, 2003, 1996, 2365, 1997, 2728, 3044, 1012, 3870, 4482, 2003, 2728,
+                                1005, 1055, 2564, 1012, 2016, 12011, 2012, 15384, 8256, 1012, 9665, 3044, 2003, 3870,
+                                1005, 1055, 2684, 1012, 2016, 2913, 2012, 15384, 4482, 102},
```

Comment:
could we potentially first use a Python script to get the input ids as per the HF tokenizer and write them into a temp file, read them from that file in this C++ file to compare with the output for the tests, and finally delete the temp files?

i think that will be a lot cleaner and will also make sure we don't miss any changed functionality in the future if HF introduces any changes which result in different input ids (e.g. from vocab/merges handling changes) and help catch such failures.

@wenbingl thoughts?

Reply:
that's ideal case, but they run these unit test on mobile platforms, which don't support that.